domain-cc/cc-app/src/python_src/api.py

```diff
@@ -47,35 +50,56 @@ def get_health_status():
     return {"status": "ok"}
 
 
+def log_lookup_table_match(
+    classification_code: int, is_in_dropdown: bool, contention_text: str
+):
+    log_as_json({"is_in_dropdown": sanitize(is_in_dropdown)})
+    log_contention_text = contention_text if is_in_dropdown else "Not in dropdown"
+
+    if classification_code:
+        already_mapped_text = contention_text.strip().lower()  # do not leak PII
+        log_as_json({"lookup_table_match": sanitize(already_mapped_text)})
+    elif is_in_dropdown:
+        log_as_json(
+            {
+                "lookup_table_match": sanitize(
+                    f"No table match for {log_contention_text}"
+                )
+            }
+        )
+    else:
+        log_as_json(
+            {"lookup_table_match": sanitize("No table match for free text entry")}
+        )
+
+
+def log_as_json(log: dict):
+    if "date" not in log.keys():
+        log.update({"date": time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())})
+    if "level" not in log.keys():
+        log.update({"level": "info"})
+    logging.info(json.dumps(log))
+
+
 @app.post("/classifier")
 def get_classification(claim: Claim) -> Optional[PredictedClassification]:
-    logging.info(
-        f"claim_id: {claim.claim_id}, form526_submission_id: {claim.form526_submission_id}"
+    log_as_json(
+        {
+            "claim_id": sanitize(claim.claim_id),
+            "form526_submission_id": sanitize(claim.form526_submission_id),
+        }
     )
     classification_code = None
     if claim.claim_type == "claim_for_increase":
-        logging.info(f"diagnostic code: {claim.diagnostic_code}")
+        log_as_json({"diagnostic code": sanitize(claim.diagnostic_code)})
         classification_code = dc_lookup_table.get(claim.diagnostic_code, None)
 
     if claim.contention_text and not classification_code:
         classification_code = dropdown_lookup_table.get(claim.contention_text, None)
         is_in_dropdown = claim.contention_text.strip().lower() in dropdown_values
-        log_contention_text = (
-            claim.contention_text if is_in_dropdown else "Not in dropdown"
+        log_lookup_table_match(
+            classification_code, is_in_dropdown, claim.contention_text
```

Comment:
nitpick: I think `is_in_dropdown` could be moved into the log function, it's only used to determine what is logging